portfolio/src/main/webapp/script.js

```diff
@@ -56,9 +56,60 @@ function showSlides(newIndex) {
 
 /** Called when website is loaded. */
 function init() {
+  showCommentForm();
   showComments();
 }
 
+/**
+ * Returns login status including current email adress and login/logout link.
+ * @return {Object} The status object.
+ */
+function getLoginStatus() {
```

Comment:
Is this function blocking or does it return a future object?
Blocking means that the function returns as soon as the full LoginStatus is there.
If it is an async method then it would return a future object which is just a handle which you can for example wait on and as soon as the information is there you can access the LoginStatus.